include/istio/utils/attributes_builder.h

```diff
@@ -99,11 +99,25 @@ class AttributesBuilder {
                        ->mutable_entries();
     entries->clear();
     for (const auto& field : struct_map.fields()) {
-      // Ignore all fields that are not string.
+      // Ignore all fields that are not string or string list.
       switch (field.second.kind_case()) {
         case google::protobuf::Value::kStringValue:
           (*entries)[field.first] = field.second.string_value();
           break;
+        case google::protobuf::Value::kListValue:
+          if (field.second.list_value().values_size() > 0) {
+            // The items in the list is converted into a
+            // comma separated string
+            std::string s;
+            for (int i = 0; i < field.second.list_value().values_size(); i++) {
```

Comment:
"int i" here is used to avoid adding a comma to the last list item. Since the for loop here of six lines is only used at other places, it is not placed in a separate util function.

Reply:
May be using string join utils under https://github.com/envoyproxy/envoy/blob/master/source/common/common/utility.h#L322

Reply:
join() input argument is a vector<std::string>: static std::string join(const std::vector<std::string>& source, const std::string& delimiter). To use join(), field.second needs to be converted to vector<std::string>. I have not found a one-liner to do the conversion.